Code/Framework/AzCore/AzCore/Module/Environment.h

```diff
@@ -352,18 +325,10 @@ namespace AZ
                 }
             }
 
-            void DestructNoLock()
-            {
-                AZ_Assert(m_isConstructed, "Variable is not constructed. Please check your logic and guard if needed!");
-                m_isConstructed = false;
-                m_moduleOwner = nullptr;
-                DestructImpl(typename AZStd::is_trivially_destructible<T>::type());
-            }
-
             void Destruct()
             {
                 AZStd::lock_guard<AZStd::spin_mutex> lock(m_mutex);
-                DestructNoLock();
+                DestructDispatchNoLock(this,DestroyTarget::Member);
```

Comment:
nitpick: Can you add a space before the DestroyTarget::Member enum?

```suggestion
                DestructDispatchNoLock(this, DestroyTarget::Member);
```

Reply:
addressed in b01d230